### ISSUE TITLE
Back-office domain reverts back to Azure default domain

### ISSUE THREAD

**Issue by robertjf** [umbraco/Umbraco.Cloud.Issues#774]
### Issue description

A website has started using the wrong url for the Umbraco Backoffice - instead of 

`https://xyz.com.au/umbraco`

 it's redirecting to 

`https://99d1f06e-abcd-42d9-94fb-1d360d29d778.azurewebsites.net/umbraco/`

I've rebooted the site temporarily fixes the problem, but it recurred after a couple of days.

The site in question is running Umbraco 10.8.6

**Comment by MichaelNielsenDK** [comment]
I'm currently experiencing the same issue. 

The site also uses Umbraco Commerce, with the QuickPay Payment Provider, and coming back from the payment gateway, it redirects to the azurewebsites.net url. 

Which is caused by it returning to an /umbraco url, and then redirects to the return url set in the settings

This fortunately can be solved by entering an url with hostnames in the return/cancel urls, then at least the user won't see the azure url. This is however not ideal, as these settings are pushed between environments, but it is a quick temporary fix. 

Umbraco v13.7.2

**Comment by MichaelNielsenDK** [comment]
We're experiencing this on yet another project. 

After being logged in with Umbraco ID,  you are redirected to the Azure url
![Image](https://github.com/user-attachments/assets/af500c0a-11b3-4580-b0fe-07403c08f9f0)

**Comment by MichaelNielsenDK** [comment]
We've just experienced this in yet another project. Is there any update on this?

**Comment by JosRaedts** [comment]
We had this issue occur today. Restarting the Umbraco platform indeed solves the issue, but it's waiting for it to happen again I feel like.

**Comment by galaxxy22** [comment]
I have seen this several times on different projects at seemingly random times. The website would redirect to the azurewebsites url. After restarting it would be fixed.

**Comment by mikkelhm** [comment]
Hello everyone - we are aware aof the issue, and it does have priority - We do also have some issues figuring out exactly what is going on here.

Can i ask you if you have a feeling about the following, just to try to narrow it down a bit.

- Is it only after visiting /umbraco that you end up being redirected to the {guid}.azurewebsites.net url ?

We had a fix for that exact issue which was released in Umbraco.Cloud.Identity.Cms version 12.2.2 -> https://www.nuget.org/packages/Umbraco.Cloud.Identity.Cms/13.2.2
And Umbraco.Cloud.Cms 15.1.4 (and above)

Before those versions what could happen is that you visit your site on your custom domain or the default umbraco.io domain on /umbraco - it would then redirect to identity.umbraco.com but with the wrong redirect url (the {guid}.azurewebsites.net) attached to it. 
This is visible in the querystring when visiting that site. It is directly where you will be redirected to after a successfull authentication at Umbraco Id. From there on you would be browsing the backoffice via the azurewebsites url

**Comment by MichaelNielsenDK** [comment]
> Is it only after visiting /umbraco that you end up being redirected to the {guid}.azurewebsites.net url ?

No, please see this related issue, where the frontend is accessible on azure urls and they are being indexed by Google
https://github.com/umbraco/Umbraco.Cloud.Issues/issues/838

**Comment by mikkelhm** [comment]
Yes we are also aware of it, but it is not the exact same issue. 
We are trying to figure out if when it happens. Really we have 3 issues, that seems related, but might not be.
 
- Redirect to the backoffice on a wrong url after authentication with Umbraco Id (this is from this thread), which should have been addressed in the Umbraco.Cloud.Identity.Cms and Umbraco.Cloud.Cms packages
- The site (frontend) generates links to azurewebsites.net - this is the one where google can end up getting those urls - We are currently looking into what happens there.
- Access to {guid}.azurewebsites.net - this has always been possible, but as long as noone links to those urls, they wont be picked up by search engines. We are also working on making those urls inaccessable from the public internet - this is what we are working on in #838 

So my question above was actually more a question if this is still happening (the redirect after authentication), even after the site has been upgraded with Umbraco.Cloud.Identity.Cms 13.2.2 or Umbraco.Cloud.Cms 15.1.4 or above.

Do you/any other have any ideas about that case ?

**Comment by MichaelNielsenDK** [comment]
> Do you/any other have any ideas about that case ?

Yes. 

The case I reported earlier, on June 12th with the screenshot, that project was using Umbraco.Cloud.Identity.Cms 13.2.4 at that time.

**Comment by mikkelhm** [comment]
Hello again

With the release of Umbraco.Cloud.Cms 13.0.1, 16.0.2 & 17.0.2, we added middleware that should prevent the site from registering that it is running on the azurewebsites.net domain at all. 

If you are not on that version, i.e. the automatic upgrade wasnt applied for one or another reason, then you should make sure the project gets upgraded with those versions. 

This shold prevent the site from spontaniously redirecting back to the azurewebsites.net domain.

If you still encounter the problem with any of those version installed, please keep this thread alive. Then we will need further investigations.

**Comment by jemayn** [comment]
As Mikkel said, this problem should no longer occur on all currently supported versions. Thus I will close this issue.

Feel free to open a new issue if the need arises.

**Comment by JosRaedts** [comment]
@mikkelhm Just a quick question: we're running on Umbraco Cloud, but we don't have the nuget package for Umbraco.Cloud.Cms installed currently. Do we need to install that package? or is it someway somehow already included without it being in the project?

**Comment by jemayn** [comment]
@JosRaedts it comes pre-installed on all Cloud sites, but as it _just_ a nuget package it is possible to uninstall it.
We assume it is there for Cloud operations, and in later versions it is the one handling for example the Umbraco ID login flow as well.

If you are on version 13+ I would install it if it is currently missing in your list of NuGet packages.
